ansible/roles/csm.rr.mgmt_nodes_placement_validation/files/mgmt_nodes_placement_validation.py

```diff
@@ -0,0 +1,257 @@
+        print(bcolors.FAIL + "Exiting placement validation...\n"  + bcolors.ENDC)
+        sys.exit(1)
+
+def validate_ceph_nodes_placement(placements_dict):
+    rack_cnt=1
+    storage_nodes_cnt=0
+    missing_cnt=0
+
+    print(bcolors.OKGREEN + bcolors.UNDERLINE + "\nValidate Management Storage (CEPH) nodes"+ bcolors.ENDC)
+
+    for rack_cnt, (rack_id, nodes) in enumerate(placements_dict.items()):
+        print("\nRack number: ", rack_cnt)
+        print("Rack ID: ", rack_id)
+        print("Management Nodes: ", nodes)
+
+        found=0
+        for node in nodes:
+            if re.match(r"^.*ncn-s00[0-9]$", node):
+                # Found storage node in a rack
+                storage_nodes_cnt+=1
+                found=1
+
+        if found == 0:
+            # Missing storage node in a rack
+            missing=1
+            missing_cnt+=1
+
+        rack_cnt+=1
+
+    print("\nTotal number of storage nodes: ", storage_nodes_cnt)
+
+    node_rack_missing_quorum=get_node_rack_missing_quorum(rack_cnt)
+
+    if storage_nodes_cnt >= min_storage_node_cnt:
+        if rack_cnt == min_rack_cnt and storage_nodes_cnt == min_master_node_cnt and  node_rack_missing_quorum >= node_rack_missing_quorum:
+            print(bcolors.FAIL + "\nOne or more racks missing storage node under minimum 3 rack condition" + bcolors.ENDC)
+            print(bcolors.FAIL + "\nPlease re arrange the storage nodes for equal distribution" + bcolors.ENDC)
+            print(bcolors.FAIL + "Exiting placement validation...\n"  + bcolors.ENDC)
+            sys.exit(1)
+        elif rack_cnt > min_rack_cnt and storage_nodes_cnt >= min_master_node_cnt and node_rack_missing_quorum > node_rack_missing_quorum:
+            print(bcolors.FAIL + "\nMore than one rack missing storage node in a", rack_cnt , "rack system" + bcolors.ENDC)
+            print(bcolors.FAIL + "\nPlease re arrange the storage nodes for equal distribution" + bcolors.ENDC)
+            print(bcolors.FAIL + "Exiting placement validation...\n"  + bcolors.ENDC)
+            sys.exit(1)
+     else:
+        print(bcolors.FAIL + "\nRack Resiliency can not be enabled with", storage_nodes_cnt, "storage nodes" + bcolors.ENDC)
+        print(bcolors.FAIL + "\nNeed minimum of 3 storage nodes configured" + bcolors.ENDC)
+        print(bcolors.FAIL + "Exiting placement validation...\n"  + bcolors.ENDC)
+        sys.exit(1)
+
+def validate_compute_uan_nodes_placement(placements_dict):
+    rack_cnt=1
+    managed_nodes_cnt=0
+    missing_cnt=0
+
+    print(bcolors.OKGREEN + bcolors.UNDERLINE + "\nValidate Managed nodes (Compute and UANs)"+ bcolors.ENDC)
+
+    for rack_cnt, (rack_id, nodes) in enumerate(placements_dict.items()):
+        print("\nRack number: ", rack_cnt)
+        print("Rack ID: ", rack_id)
+        print("Management/ Managed Nodes: ", nodes)
+
+        found=0
+        for node in nodes:
+            if re.match(r"^.*nid00000[0-9]$", node) or re.match(r"^.*uan0[0-9]$", node):
```

Comment:
This pattern looks like it will only match nid0-9 or uan0-9. There definitely can be more than that many compute nodes, and I am not aware of any limit on the number of uans.